GUI-pause-menu/Assets/Scripts/PauseMenu.cs

```diff
@@ -0,0 +1,186 @@
+﻿using System;
+using System.Linq;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UIElements;
+using Unity.UIElements.Runtime;
+
+namespace GUI
+{
+    public class PauseMenu : MonoBehaviour
+    {
+        /// <summary>
+        ///     Uxml/Uss ui element class selectors.
+        ///     Match strings with Uxml class names and types.
+        ///     
+        ///     C_MENU_BUTTON type = Button
+        ///     C_CLOSE_BUTTON type = Button
+        ///     C_SCREEN_CONTAINER type = VisualElement
+        ///     TEMPLATE_CONTAINER type = TemplateContainer (embedded uxml)
+        ///     DEFAULT_SCREEN = VisualElement
+        /// </summary>
+        public const string C_MENU_BUTTON = "menu-button";              
+        public const string C_CLOSE_BUTTON = "close-button";
+        public const string C_SCREEN_CONTAINER = "screen-container";
+        public const string TEMPLATE_CONTAINER = "screen-template";
+        public const string DEFAULT_SCREEN = "screen-I";
+
+        public Button closeButton;
+
+        public VisualElement currentScreen;
+
+        public List<Button> menuButtons = new List<Button>();
+
+        public List<VisualElement> screenContainers = new List<VisualElement>();
+
+        // Store screen templates for targeting 
+        public List<TemplateContainer> screens = new List<TemplateContainer>();
+
+        // Panel renderer (display UI in runtime scene) 
+        private PanelRenderer pr;
+
+        // Virtual Tree (root) for querys
+        private VisualElement vt;
+
+        void Start()
+        {
+            // Set panel renderer values
+            pr = GetComponent<PanelRenderer>();
+            vt = pr.visualTree;
+
+            // Set main UI elements
+            try
+            {
+                closeButton = vt.Query<Button>(C_CLOSE_BUTTON).First();
+                menuButtons = vt.Query<Button>(classes: C_MENU_BUTTON).ToList();
+                screenContainers = vt.Query<VisualElement>(classes: C_SCREEN_CONTAINER).ToList();
+
+                // TODO: Hard coded, should change based on userinput (key "i" opens "inventory", "c" "character", etc.) 
+                // See: https://trello.com/c/wUNnTvBN/18-p-rebind-rebind-controls
+                currentScreen = vt.Query<VisualElement>(DEFAULT_SCREEN).First();
```

Comment:
Misschien handig om in Trello een nieuwe kaart aan te maken voor deze TODO wanneer deze branch gemerged is. Anders wordt het misschien vergeten.

Reply:
https://trello.com/c/z4tsq8aW/59-pause-menu-ui-controls-toevoegen